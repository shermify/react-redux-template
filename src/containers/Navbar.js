import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <div>
    <nav className="navbar navbar-toggleable-md navbar-inverse bg-primary">
      <button
        className="navbar-toggler navbar-toggler-right"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNavDropdown"
        aria-controls="navbarNavDropdown"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <Link to="/" className="navbar-brand">React-Redux Template</Link>
      <div className="collapse navbar-collapse" id="navbarNavDropdown">
        <ul className="navbar-nav">
          <li className="nav-item active">
            <Link to="/" className="nav-link">Home <span className="sr-only">(current)</span></Link>
          </li>
          <li className="nav-item">
            <Link className="nav-link" to="/example">Example Page</Link>
          </li>
        </ul>
      </div>
    </nav>
  </div>
);

export default Navbar;
