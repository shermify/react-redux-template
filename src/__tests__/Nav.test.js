// Link.react-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../containers/Navbar';

it('renders correctly', () => {
  const tree = renderer
    .create(<MemoryRouter><Navbar /></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
