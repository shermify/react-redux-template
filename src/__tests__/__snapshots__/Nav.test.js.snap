// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-toggleable-md navbar-inverse bg-primary"
  >
    <button
      aria-controls="navbarNavDropdown"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbarNavDropdown"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      React-Redux Template
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavDropdown"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home 
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/example"
            onClick={[Function]}
          >
            Example Page
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
